webrtc/RTCSctpTransport-constructor.html

```diff
@@ -5,85 +5,121 @@
-        'Expect sctp.maxMessageSize to be a number');
-    });
-  }, 'setRemoteDescription() with answer containing data media should initialize pc.sctp');
-
-  promise_test(t => {
-    const pc = new RTCPeerConnection();
-    t.add_cleanup(() => pc.close());
-    assert_equals(pc.sctp, null);
-
-    return generateDataChannelOffer(pc)
-    .then(offer => pc.setRemoteDescription(offer))
-    .then(() => pc.createAnswer())
-    .then(answer => pc.setLocalDescription(answer))
-    .then(() => {
-      const { sctp } = pc;
-      assert_not_equals(sctp, null);
-      assert_true(sctp instanceof RTCSctpTransport,
-        'Expect pc.sctp to be instance of RTCSctpTransport');
-
-      assert_true(sctp.transport instanceof RTCDtlsTransport,
-        'Expect sctp.transport to be instance of RTCDtlsTransport');
-
-      assert_true(typeof sctp.maxMessageSize == 'number',
-        'Expect sctp.maxMessageSize to be a number');
-    });
-  }, 'setLocalDescription() with answer containing data media should initialize pc.sctp');
+'use strict';
+
+// Test is based on the following revision:
+// https://rawgit.com/w3c/webrtc-pc/1cc5bfc3ff18741033d804c4a71f7891242fb5b3/webrtc.html
+
+// The following helper functions are called from RTCPeerConnection-helper.js:
+//   generateDataChannelOffer()
+//   generateAnswer()
+
+/*
+  6.1.
+
+    partial interface RTCPeerConnection {
+      readonly attribute RTCSctpTransport? sctp;
+      ...
+    };
+
+  6.1.1.
+
+    interface RTCSctpTransport {
+        readonly attribute RTCDtlsTransport      transport;
+        readonly attribute RTCSctpTransportState state;
+        readonly attribute unrestricted double   maxMessageSize;
+                 attribute EventHandler          onstatechange;
+    };
+
+  4.4.1.1.  Constructor
+    9.      Let connection have an [[SctpTransport]] internal slot, initialized to null.
+
+  4.4.1.6.  Set the RTCSessionSessionDescription
+    2.2.6.  If description is of type "answer" or "pranswer", then run the
+            following steps:
+      1.    If description initiates the establishment of a new SCTP association, as defined in
+            [SCTP-SDP], Sections 10.3 and 10.4, create an RTCSctpTransport with an initial state
+            of "connecting" and assign the result to the [[SctpTransport]] slot.
+ */
+
+promise_test(async (t) => {
+  const pc1 = new RTCPeerConnection();
+  const pc2 = new RTCPeerConnection();
+  t.add_cleanup(() => pc1.close());
+  t.add_cleanup(() => pc2.close());
+
+  assert_equals(pc1.sctp, null, 'RTCSctpTransport must be null');
+
+  const offer = await generateDataChannelOffer(pc1);
+  await Promise.all([pc1.setLocalDescription(offer), pc2.setRemoteDescription(offer)]);
+  const answer = await pc2.createAnswer();
+  await pc1.setRemoteDescription(answer);
+
+  assert_equals(pc1.sctp, null, 'RTCSctpTransport must remain null');
+}, 'setRemoteDescription() with answer not containing data media should not initialize pc.sctp');
```

Comment:
This can't possibly be right. If you have a datachannel offer, and answer with default parameters, you should get a datachannel established, and the SctpTransport along with it.
In order to get this result, you would need to edit the SCTP section out of the generated answer.

Reply:
These tests have been altered in the refactoring process. They initially used audio: https://github.com/web-platform-tests/wpt/blob/99d792a605af57c390a39f7f817dad8135d0c7d2/webrtc/RTCSctpTransport-constructor.html#L45:L75

Reply:
Fixed, PTAL